File: app/src/main/java/com/xyoye/dandanplay/ui/activities/SplashActivity.java
package com.xyoye.dandanplay.ui.activities;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.view.KeyEvent;
import android.view.WindowManager;
import android.view.animation.AlphaAnimation;
import android.view.animation.Animation;
import android.view.animation.LinearInterpolator;
import android.widget.LinearLayout;

import com.xyoye.dandanplay.R;
import com.xyoye.dandanplay.base.BaseMvpActivity;
import com.xyoye.dandanplay.mvp.impl.SplashPresenterImpl;
import com.xyoye.dandanplay.mvp.presenter.SplashPresenter;
import com.xyoye.dandanplay.mvp.view.SplashView;
import com.xyoye.dandanplay.ui.weight.anim.path.TextPathAnimView;
import com.xyoye.dandanplay.ui.weight.anim.svg.AnimatedSvgView;
import com.xyoye.dandanplay.utils.AppConfig;

import butterknife.BindView;

/**
 * Created by xyoye on 2018/7/15.
 */

public class SplashActivity extends BaseMvpActivity<SplashPresenter> implements SplashView {

    @BindView(R.id.icon_svg_view)
    AnimatedSvgView iconSvgView;
    @BindView(R.id.text_path_view)
    TextPathAnimView textPathView;
    @BindView(R.id.address_ll)
    LinearLayout addressLl;


    @Override
    protected void process(Bundle savedInstanceState) {
        super.process(savedInstanceState);
    }

    @Override
    protected int initPageLayoutID() {
        return R.layout.activity_splash;
    }

    @Override
    public void initView() {
        presenter.checkToken();

        //是否关闭启动页
        if (AppConfig.getInstance().isCloseSplashPage()) {
            launchActivity();
            return;
        }

        getWindow().addFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
        getWindow().addFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN);
        getWindow().addFlags(WindowManager.LayoutParams.FLAG_SHOW_WALLPAPER);

        AlphaAnimation alphaAnimation = new AlphaAnimation(0, 1);
        alphaAnimation.setRepeatCount(Animation.ABSOLUTE);
        alphaAnimation.setInterpolator(new LinearInterpolator());
        alphaAnimation.setDuration(2000);

        textPathView.setAnimListener(new TextPathAnimView.AnimListener() {
            @Override
            public void onStart() {

            }

            @Override
            public void onEnd() {
                textPathView.postDelayed(() -> launchActivity(), 350);
            }

            @Override
            public void onLoop() {

            }
        });

        textPathView.startAnim();
        iconSvgView.start();
        addressLl.startAnimation(alphaAnimation);
    }

    @Override
    public void initListener() {

    }

    @NonNull
    @Override
    protected SplashPresenter initPresenter() {
        return new SplashPresenterImpl(this, this);
    }

    @Override
    public void onBackPressed() {

    }

    @Override
    public boolean dispatchKeyEvent(KeyEvent event) {
        if (event.getKeyCode() == KeyEvent.KEYCODE_BACK) {
            return true;
        } else {
            return super.dispatchKeyEvent(event);
        }
    }

    private void launchActivity() {
        Intent intent = new Intent(SplashActivity.this, MainActivity.class);
        startActivity(intent);
        overridePendingTransition(R.anim.anim_activity_enter, R.anim.anim_activity_exit);
        this.finish();
    }
}

File: app/src/main/java/com/xyoye/dandanplay/ui/splash/SplashActivity.kt
package com.xyoye.dandanplay.ui.splash

import android.content.Intent
import android.graphics.Color
import android.os.Build
import android.view.KeyEvent
import android.view.WindowInsets
import android.view.WindowManager
import android.view.animation.AlphaAnimation
import android.view.animation.Animation
import android.view.animation.LinearInterpolator
import androidx.appcompat.app.AppCompatDelegate
import com.gyf.immersionbar.ImmersionBar
import com.xyoye.common_component.base.BaseAppCompatActivity
import com.xyoye.common_component.config.AppConfig
import com.xyoye.dandanplay.R
import com.xyoye.dandanplay.databinding.ActivitySplashBinding
import com.xyoye.dandanplay.ui.main.MainActivity
import com.xyoye.dandanplay.utils.image_anim.path.TextPathAnimView

/**
 * Created by xyoye on 2020/7/27.
 */

class SplashActivity : BaseAppCompatActivity<ActivitySplashBinding>() {
    override fun getLayoutId() = R.layout.activity_splash

    override fun initStatusBar() {
        ImmersionBar.with(this)
            .transparentBar()
            .statusBarDarkFont(false)
            .init()
    }

    override fun initView() {
        window.run {
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
                insetsController?.hide(WindowInsets.Type.statusBars())
            } else {
                addFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN)
            }
            statusBarColor = Color.TRANSPARENT
            addFlags(WindowManager.LayoutParams.FLAG_SHOW_WALLPAPER)
        }

        val mode = AppConfig.getDarkMode()
        AppCompatDelegate.setDefaultNightMode(mode)

        if (!AppConfig.getShowSplashAnimation()) {
            launchActivity()
            return
        }

        val alphaAnimation = AlphaAnimation(0f, 1f)
        alphaAnimation.apply {
            repeatCount = Animation.ABSOLUTE
            interpolator = LinearInterpolator()
            duration = 2000
        }

        val appName = "弹弹play 概念版 v${packageManager.getPackageInfo(packageName, 0).versionName}"

        dataBinding.run {
            appNameTv.text = appName

            textPathView.setAnimListener(object : TextPathAnimView.AnimListener {
                override fun onStart() {

                }

                override fun onEnd() {
                    textPathView.postDelayed({
                        launchActivity()
                    }, 350)
                }

                override fun onLoop() {

                }
            })
            textPathView.startAnim()
            iconSvgView.start()
            appNameLl.startAnimation(alphaAnimation)
        }
    }

    override fun dispatchKeyEvent(event: KeyEvent?): Boolean {
        return if (event!!.keyCode == KeyEvent.KEYCODE_BACK) {
            true
        } else {
            super.dispatchKeyEvent(event)
        }
    }

    override fun onDestroy() {
        dataBinding.textPathView.cancelAnim()
        super.onDestroy()
    }

    private fun launchActivity() {
        val intent = Intent(this@SplashActivity, MainActivity::class.java)
        startActivity(intent)
        overridePendingTransition(
            com.xyoye.common_component.R.anim.anim_activity_enter,
            com.xyoye.common_component.R.anim.anim_activity_exit
        )
        finish()
    }
}